### PR TITLE
LOOP-1400: Add feature flag for delete all button 

### PR DESCRIPTION
**Pull request by novalegra**
https://tidepool.atlassian.net/browse/LOOP-1400

### CODE REVIEW

File: Common/FeatureFlags.swift [hunk 1]
```diff
@@ -90,13 +91,18 @@ struct FeatureFlagConfiguration: Decodable {
...
-        
-        // Swift compiler config is inverse, since the default state is enabled.
+
         #if MOCK_THERAPY_SETTINGS_ENABLED
         self.mockTherapySettingsEnabled = true
         #else
         self.mockTherapySettingsEnabled = false
         #endif
+        
+        #if DELETE_ALL_BUTTON_DISABLED
```
Comment (darinkrauss) [inline Common/FeatureFlags.swift:102]: [nit] This is one that needs the comment deleted above. ;)

File: Common/FeatureFlags.swift [hunk 2]
```diff
@@ -22,6 +22,7 @@ struct FeatureFlagConfiguration: Decodable {
     let observeHealthKitSamplesFromOtherApps: Bool
     let includeServicesInSettingsEnabled: Bool
     let mockTherapySettingsEnabled: Bool
+    let deleteAllButtonDisabled: Bool
```
Comment (darinkrauss) [inline Common/FeatureFlags.swift]: All other feature flags have the "Enabled" suffix. Please flip this logic.

File: Common/FeatureFlags.swift (hunk 1 above)
Comment (darinkrauss) [reply Common/FeatureFlags.swift:102]: Once you flip the logic as mentioned above.

File: Loop/View Controllers/StatusTableViewController.swift [hunk 3]
```diff
@@ -1115,6 +1115,7 @@ final class StatusTableViewController: LoopChartsTableViewController {
         case let vc as InsulinDeliveryTableViewController:
             vc.doseStore = deviceManager.doseStore
             vc.hidesBottomBarWhenPushed = true
+            vc.disableDeleteAllButton = FeatureFlags.deleteAllButtonDisabled
```
Comment (darinkrauss) [inline Loop/View Controllers/StatusTableViewController.swift]: You'll need to invert this, as well.

**Review by darinkrauss — CHANGES_REQUESTED** [review]
To be consistent, I suggest changing "Disabled" to "Enabled" and inverting the logic.

File: Common/FeatureFlags.swift [hunk 4]
```diff
@@ -90,13 +91,18 @@ struct FeatureFlagConfiguration: Decodable {
...
-        
-        // Swift compiler config is inverse, since the default state is enabled.
+
         #if MOCK_THERAPY_SETTINGS_ENABLED
         self.mockTherapySettingsEnabled = true
         #else
         self.mockTherapySettingsEnabled = false
         #endif
+        
+        #if DELETE_ALL_BUTTON_ENABLED
```
Comment (darinkrauss) [inline Common/FeatureFlags.swift]: Sorry, I guess I wasn't clear here. The default behavior should be if the environment variable is NOT defined (for DIY Loop). Therefore, since we want to disable the Delete All button in Tidepool Loop, the environment variable needs to be called `DELETE_ALL_BUTTON_DISABLED` while the property is `deleteAllButtonEnabled`.

Think about it this way, for DIY Loop, *none* of these environment variables will be defined. Therefore, with the current implementation `deleteAllButtonEnabled` will be `false`.

So, in short, rename `DELETE_ALL_BUTTON_ENABLED` to `DELETE_ALL_BUTTON_DISABLED`, leave `deleteAllButtonEnabled` named the same, but reverse the `true` and `false` below. Finally, add the "// Swift compiler config is inverse, since the default state is enabled." comment above this line.

**Review by darinkrauss — CHANGES_REQUESTED** [review]
Sorry, I wasn't clear. One comment to change.

File: Common/FeatureFlags.swift (hunk 1 above)
Comment (darinkrauss) [reply Common/FeatureFlags.swift:102]: Last thing, I promise. Just add the "// Swift compiler config is inverse, since the default state is enabled." above this line. :)

**Review by darinkrauss — CHANGES_REQUESTED** [review]
Sorry, one tiny last change for completeness.

**Review by darinkrauss — APPROVED** [review]
LGTM! 👍 (Thanks for putting up with all of my requests!)
